main.tf

```diff
@@ -81,7 +82,7 @@ data "aws_iam_policy_document" "default" {
     effect = "Allow"
 
     resources = [
-      "arn:aws:route53:::hostedzone/${data.aws_route53_zone.default.zone_id}",
+      "${formatlist("arn:aws:route53:::hostedzone/%s", data.aws_route53_zone.default.*.zone_id)}",
```

Comment:
`formatlist` returns a list, but we have already wrapped it in a list. Will TF flatten it and not use list of list?

Reply:
Someone flattens it, not sure who. This definitely works, though.